hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/monitoring/MonitoringCache.java

```diff
@@ -0,0 +1,164 @@
+package pl.allegro.tech.hermes.management.infrastructure.monitoring;
+
+import com.google.common.collect.Lists;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.api.TopicName;
+import pl.allegro.tech.hermes.common.kafka.KafkaNamesMapper;
+import pl.allegro.tech.hermes.management.config.MonitoringProperties;
+import pl.allegro.tech.hermes.management.config.kafka.KafkaClustersProperties;
+import pl.allegro.tech.hermes.management.config.kafka.KafkaNamesMappers;
+import pl.allegro.tech.hermes.management.config.kafka.KafkaProperties;
+import pl.allegro.tech.hermes.management.domain.subscription.SubscriptionService;
+import pl.allegro.tech.hermes.management.domain.topic.TopicService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.kafka.clients.CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL;
+import static org.apache.kafka.clients.CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG;
+import static org.apache.kafka.clients.CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_JAAS_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_MECHANISM;
+import static pl.allegro.tech.hermes.api.TopicName.fromQualifiedName;
+
+public class MonitoringCache {
+
+    private static final Logger logger = LoggerFactory.getLogger(MonitoringCache.class);
+
+    private final KafkaClustersProperties kafkaClustersProperties;
+    private final KafkaNamesMappers kafkaNamesMappers;
+    private final MonitoringProperties monitoringProperties;
+    private final SubscriptionService subscriptionService;
+    private final TopicService topicService;
+
+    private volatile List<List<TopicAndSubscription>> readySubscriptionsWithUnassignedPartitionsCache = new ArrayList<>();
```

Comment:
The volatile on collection modified by changing its content does have no effect. https://javarevisited.blogspot.com/2017/01/can-we-make-array-volatile-in-java.html#axzz7tZoA4Fz9
It makes sense to use volatile if you will change the reference